pkg/provider/github/app/token.go

```diff
@@ -50,31 +47,27 @@ func (ip *Install) GetAndUpdateInstallationID(ctx context.Context) (string, stri
 		return "", "", 0, err
 	}
 
-	installationURL := *ip.ghClient.APIURL + keys.InstallationURL
+	apiURL := *ip.ghClient.APIURL
 	enterpriseHost = ip.request.Header.Get("X-GitHub-Enterprise-Host")
 	if enterpriseHost != "" {
 		// NOTE: Hopefully this works even when the ghe URL is on another host than the api URL
-		installationURL = "https://" + enterpriseHost + "/api/v3" + keys.InstallationURL
+		apiURL = "https://" + enterpriseHost + "/api/v3"
 	}
 
-	res, err := GetReponse(ctx, http.MethodGet, installationURL, jwtToken, ip.run)
-	if err != nil {
-		return "", "", 0, err
-	}
-
-	if res.StatusCode >= 300 {
-		return "", "", 0, fmt.Errorf("Non-OK HTTP status while getting installation URL: %s : %d", installationURL, res.StatusCode)
-	}
-
-	defer res.Body.Close()
-	data, err := io.ReadAll(res.Body)
-	if err != nil {
-		return "", "", 0, err
-	}
-
-	installationData := []gt.Installation{}
-	if err = json.Unmarshal(data, &installationData); err != nil {
-		return "", "", 0, err
+	logger := logging.FromContext(ctx)
+	opt := &gt.ListOptions{PerPage: ip.ghClient.PaginedNumber}
+	client, _, _ := github.MakeClient(ctx, apiURL, jwtToken)
```

Comment:
Can we handle error here ??

Reply:
MakeClient does not return error, neither the go-github library NewClient does. I skimmed through its code and it simply initiates a client. I can check for a nil value client, but it looks like that value can't be nil either.